src/Sentry.AspNetCore/SentryWebHostBuilderExtensions.cs

```diff
@@ -95,14 +95,15 @@ public static IWebHostBuilder UseSentry(
             _ = logging.Services.AddSingleton<ILoggerProvider, SentryAspNetCoreLoggerProvider>();
             _ = logging.Services.AddSingleton<ILoggerProvider, SentryAspNetCoreStructuredLoggerProvider>();
 
-            _ = logging.AddFilter<SentryAspNetCoreLoggerProvider>(
-                "Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware",
-                LogLevel.None);
-            _ = logging.AddFilter<SentryAspNetCoreStructuredLoggerProvider>(static (string? categoryName, LogLevel logLevel) =>
+            // Add a delegate rule in order to ignore Configuration like "appsettings.json" and "appsettings.{HostEnvironment}.json"
+            _ = logging.AddFilter<SentryAspNetCoreLoggerProvider>(static (string? categoryName, LogLevel logLevel) =>
             {
                 return categoryName is null
-                    || (categoryName != "Sentry.ISentryClient" && categoryName != "Sentry.AspNetCore.SentryMiddleware");
+                       || categoryName != "Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware";
             });
+            // Add non-delegate rules in order to respect Configuration like "appsettings.json" and "appsettings.{HostEnvironment}.json"
+            _ = logging.AddFilter<SentryAspNetCoreStructuredLoggerProvider>("Sentry.ISentryClient", LogLevel.None);
```

Comment:
```suggestion
            _ = logging.AddFilter<SentryAspNetCoreStructuredLoggerProvider>(nameof(Sentry.ISentryClient), LogLevel.None);
```

I assume these are designed to work with `ILogger<T>` in which case the category name essentially has to match a type name. If we use `nameof` then we guard against this breaking if any of the types ever get renamed.

Same again for the other instances where we've done this in other files.

Reply:
This would not work.
The Category is the Full-Type-Name of the generic `TCategoryName`.
But `nameof(Sentry.ISentryClient)` does not include the _namespace_ and results in `"ISentryClient"`, rather than `"Sentry.ISentryClient"`.

Instead, we would need to do `typeof(ISentryClient).FullName` ... but that comes along with (a very small) cost at runtime, compared to the interned string literal ... but this is what we actually do in the unit-tests:
https://github.com/getsentry/sentry-dotnet/blob/ce4b5e121b2e8fcce365692e118a9caddff10489/test/Sentry.AspNetCore.Tests/SentryWebHostBuilderExtensionsTests.cs#L130

I do see the argument about refactoring safety, though!

The options that I see are:
1. keep as is
   - since all respective tests use the refactoring-safe `typeof(TCategoryName).FullName`
   - so these tests would fail if we had a typo in the `AddFilter` invocation, or would move/rename the type
2. change to `typeof(TCategoryName).FullName`
   - this does come with a very small cost at runtime
   - but to be fair, it's only called once one during Startup of the Host application, so this is quite neglectable
3. convert to an `internal const string` on `TCategoryName`
   - it would be just below the _type-declaration_ to become more obvious when we would move/change the type
   - and we would only need to change it at a single location
   - still not truly refactoring-safe
4. convert to an `internal static readonly string` on on `TCategoryName`, initialized to `typeof(TCategoryName).FullName`
   - a bit of a combination of above arguments